### PR TITLE
sql/schema: simplify the ExecQuerier interface.

### PR DESCRIPTION
**Pull request by a8m**
Make it simpler to implement it by tools like Ent

### CODE REVIEW

File: sql/postgres/inspect.go [hunk 1]
```diff
@@ -121,9 +125,14 @@ func (i *inspect) table(ctx context.Context, name string, opts *schema.InspectTa
 		query = tableSchemaQuery
 		args = append(args, opts.Schema)
 	}
-	row := i.QueryRowContext(ctx, query, args...)
-	var tSchema, comment sql.NullString
-	if err := row.Scan(&tSchema, &comment); err != nil {
+	var (
+		tSchema, comment sql.NullString
+		rows, err        = i.QueryContext(ctx, query, args...)
```
Comment (yonidavidson) [inline sql/postgres/inspect.go:130]: is this correct formatting?

**Review by yonidavidson — APPROVED** [review]

File: sql/schema/inspect.go [hunk 2]
```diff
@@ -27,11 +27,8 @@ func IsNotExistError(err error) bool {
 	return errors.As(err, &e)
 }
 
-// ExecQuerier wraps the standard sql.DB methods.
+// ExecQuerier wraps the 2 standard sql.DB methods.
```
Comment (masseelch) [inline sql/schema/inspect.go:30]:
```suggestion
// ExecQuerier wraps the two standard sql.DB methods.
```

File: sql/internal/sqlx/sqlx.go [hunk 3]
```diff
@@ -20,6 +20,18 @@ func ValidString(s sql.NullString) bool {
 	return s.Valid && s.String != "" && strings.ToLower(s.String) != "null"
 }
 
+// ScanOne scans one record and closes the rows at the end.
+func ScanOne(rows *sql.Rows, dest ...interface{}) error {
```
Comment (masseelch) [inline sql/internal/sqlx/sqlx.go:24]: This should error out if there are more than 1 row, should't it?

File: sql/postgres/inspect.go (hunk 1 above)
Comment (a8m) [reply sql/postgres/inspect.go:130]: Yes.

File: sql/internal/sqlx/sqlx.go (hunk 3 above)
Comment (a8m) [reply sql/internal/sqlx/sqlx.go:24]: I tried to keep the same behavior as `sql.Row`.

Comment (masseelch) [reply sql/internal/sqlx/sqlx.go:24]: I was thinking about [database/sql](https://pkg.go.dev/database/sql#DB.QueryRow). It errors if there is more than one row in the result set.

Comment (masseelch) [reply sql/internal/sqlx/sqlx.go:24]: Sorry, it behaves exactly like you said.

**Pull request merged**